bin/improver-wxcode

```diff
@@ -99,27 +104,39 @@ def main():
     the weather symbols plugin.
     """
 
-    diagnostics = interrogate_decision_tree()
+    diagnostics = interrogate_decision_tree('uk')
     n_files = len(diagnostics)
     dlist = (' - {}\n'*n_files)
 
+    diagnostics_global = interrogate_decision_tree('global')
+    n_files_global = len(diagnostics_global)
+    dlist_global = (' - {}\n'*n_files_global)
+
     parser = ArgParser(
         description='Calculate gridded weather symbol codes.\nThis plugin '
         'requires a specific set of input diagnostics, where data\nmay be in '
         'any units to which the thresholds given below can\nbe converted:\n' +
-        dlist.format(*diagnostics),
+        dlist.format(*diagnostics)
+        + '\n\n or for global data\n\n' +
+        dlist_global.format(*diagnostics_global),
         formatter_class=RawTextHelpFormatter)
 
     parser.add_argument(
-        'input_filepaths', metavar='INPUT_FILES', nargs=n_files,
+        'input_filepaths', metavar='INPUT_FILES', nargs="+",
         help='Paths to files containing the required input diagnostics.')
     parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
                         help='The output path for the processed NetCDF.')
+    parser.add_argument("--wxtree", metavar="WXTREE",
+                        default="uk",
+                        choices=["uk", "global"],
+                        help="Weather Code tree"
```

Comment:
Sorry I don't agree the different trees have nothing to do with convection but are due to one being high resolution over the UK and one being lower resolution Lat/Lon global data. 

In the notebook I show that we can do vicinity processing on the lat/lon grid but it is slow. The main question is do we need it on the lower resolution model which is not resolving individual rain showers.

Reply:
I will leave this for a second reviewer to weigh in on. I'm still a little concerned about the geographic specificity in the naming.

Reply:
Having discussed: we think "global" is OK, but it would be preferable to replace "uk" with "local" or "high_resolution", to reflect the logic for vicinity processing.

Reply:
Changed to high_resolution